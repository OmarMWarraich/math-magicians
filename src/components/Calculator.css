
.calculator {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 40%;
  height: 100%;
  margin: 200px auto;
  background-color: #f2f2f2;
  border: 0.5px solid #ccc;
}

.calculator-display {
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: right;
  width: 100%;
  height: 1.5rem;
  background-color: #808080;
}

.calculator-keypad {
  display: grid;
  grid-template-columns: auto auto auto;
  grid-auto-flow: dense;
  height: 60%;
  width: 75%;
}

.calculator-keypad * {
  background-color: #f8f8ff;
  border-color: #fff;
}

.key-clear,
.key-percent {
  font-size: 9px;
}

.key-clear,
.key-sign,
.key-percent {
  grid-row: 1;
}

.key-9,
.key-8,
.key-7 {
  grid-row: 2;
}

.key-6,
.key-5,
.key-4 {
  grid-row: 3;
}

.key-3,
.key-2,
.key-1 {
  grid-row: 4;
}

.key-0,
.key-dot {
  grid-row: 5;
}

.key-0 {
  grid-column: 1 / span 2;
}

.container {
  display: flex;
  width: 100%;
}

.operator-keys {
  display: flex;
  flex-direction: column;
  width: 25%;
}

.operator-keys * {
  background-color: orange;
  border-color: #fff;
}
