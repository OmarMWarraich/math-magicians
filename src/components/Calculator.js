import React, { Component } from 'react';
import './Calculator.css';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends Component {
  render() {
    return (
      <div className="calculator">
        <div className="calculator-display">0</div>
        <div className="container">
          <div className="calculator-keypad">
            <button type="button" className="calculator-key key-clear">AC</button>
            <button type="button" className="calculator-key key-sign">±</button>
            <button type="button" className="calculator-key key-percent">%</button>
            <button type="button" className="calculator-key key-0">0</button>
            <button type="button" className="calculator-key key-dot">●</button>
            <button type="button" className="calculator-key key-1">1</button>
            <button type="button" className="calculator-key key-2">2</button>
            <button type="button" className="calculator-key key-3">3</button>
            <button type="button" className="calculator-key key-4">4</button>
            <button type="button" className="calculator-key key-5">5</button>
            <button type="button" className="calculator-key key-6">6</button>
            <button type="button" className="calculator-key key-7">7</button>
            <button type="button" className="calculator-key key-8">8</button>
            <button type="button" className="calculator-key key-9">9</button>
          </div>
          <div className="operator-keys">
            <button type="button" className="calculator-key key-divide">÷</button>
            <button type="button" className="calculator-key key-multiply">×</button>
            <button type="button" className="calculator-key key-subtract">−</button>
            <button type="button" className="calculator-key key-add">+</button>
            <button type="button" className="calculator-key key-equals">=</button>
          </div>
        </div>

      </div>
    );
  }
}

export default Calculator;
