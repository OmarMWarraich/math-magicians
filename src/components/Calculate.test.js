import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Calculator from './Calculator';

describe('test Calculator component', () => {
  test('renders Calculator component', () => {
    const component = render(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
