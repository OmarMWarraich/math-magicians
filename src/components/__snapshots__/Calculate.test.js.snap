// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Calculator component renders Calculator component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="title"
        >
          <h1>
            Let's do some math!
          </h1>
        </div>
        <div
          class="calculator"
        >
          <div
            class="calculator-display"
          >
            0
          </div>
          <div
            class="container"
          >
            <div
              class="calculator-keypad"
            >
              <button
                class="calculator-key key-clear"
                type="button"
              >
                AC
              </button>
              <button
                class="calculator-key key-sign"
                type="button"
              >
                +/-
              </button>
              <button
                class="calculator-key key-percent"
                type="button"
              >
                %
              </button>
              <button
                class="calculator-key key-0"
                type="button"
              >
                0
              </button>
              <button
                class="calculator-key key-dot"
                type="button"
              >
                .
              </button>
              <button
                class="calculator-key key-1"
                type="button"
              >
                1
              </button>
              <button
                class="calculator-key key-2"
                type="button"
              >
                2
              </button>
              <button
                class="calculator-key key-3"
                type="button"
              >
                3
              </button>
              <button
                class="calculator-key key-4"
                type="button"
              >
                4
              </button>
              <button
                class="calculator-key key-5"
                type="button"
              >
                5
              </button>
              <button
                class="calculator-key key-6"
                type="button"
              >
                6
              </button>
              <button
                class="calculator-key key-7"
                type="button"
              >
                7
              </button>
              <button
                class="calculator-key key-8"
                type="button"
              >
                8
              </button>
              <button
                class="calculator-key key-9"
                type="button"
              >
                9
              </button>
            </div>
            <div
              class="operator-keys"
            >
              <button
                class="calculator-key key-divide"
                type="button"
              >
                รท
              </button>
              <button
                class="calculator-key key-multiply"
                type="button"
              >
                x
              </button>
              <button
                class="calculator-key key-subtract"
                type="button"
              >
                -
              </button>
              <button
                class="calculator-key key-add"
                type="button"
              >
                +
              </button>
              <button
                class="calculator-key key-equals"
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="title"
      >
        <h1>
          Let's do some math!
        </h1>
      </div>
      <div
        class="calculator"
      >
        <div
          class="calculator-display"
        >
          0
        </div>
        <div
          class="container"
        >
          <div
            class="calculator-keypad"
          >
            <button
              class="calculator-key key-clear"
              type="button"
            >
              AC
            </button>
            <button
              class="calculator-key key-sign"
              type="button"
            >
              +/-
            </button>
            <button
              class="calculator-key key-percent"
              type="button"
            >
              %
            </button>
            <button
              class="calculator-key key-0"
              type="button"
            >
              0
            </button>
            <button
              class="calculator-key key-dot"
              type="button"
            >
              .
            </button>
            <button
              class="calculator-key key-1"
              type="button"
            >
              1
            </button>
            <button
              class="calculator-key key-2"
              type="button"
            >
              2
            </button>
            <button
              class="calculator-key key-3"
              type="button"
            >
              3
            </button>
            <button
              class="calculator-key key-4"
              type="button"
            >
              4
            </button>
            <button
              class="calculator-key key-5"
              type="button"
            >
              5
            </button>
            <button
              class="calculator-key key-6"
              type="button"
            >
              6
            </button>
            <button
              class="calculator-key key-7"
              type="button"
            >
              7
            </button>
            <button
              class="calculator-key key-8"
              type="button"
            >
              8
            </button>
            <button
              class="calculator-key key-9"
              type="button"
            >
              9
            </button>
          </div>
          <div
            class="operator-keys"
          >
            <button
              class="calculator-key key-divide"
              type="button"
            >
              รท
            </button>
            <button
              class="calculator-key key-multiply"
              type="button"
            >
              x
            </button>
            <button
              class="calculator-key key-subtract"
              type="button"
            >
              -
            </button>
            <button
              class="calculator-key key-add"
              type="button"
            >
              +
            </button>
            <button
              class="calculator-key key-equals"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
